Synopsis.app crashes if -temp file already exists

<filename>-temp.mov needs to be deleted before starting or synopsis crashes with the following logs/stacktrace:

```
2017-01-15 16:45:49.355114 Synopsis[9426:113691]  [ERROR] Unable to start transcode:
2017-01-15 16:45:49.355117 Synopsis[9426:113504]  [ERROR] Unable to start transcode:
2017-01-15 16:45:49.355280 Synopsis[9426:113691]  [ERROR] Write ErrorError Domain=AVFoundationErrorDomain Code=-11823 "Cannot Save" UserInfo={NSUnderlyingError=0x618000e55c60 {Error Domain=NSOSStatusErrorDomain Code=-12412 "(null)"}, NSLocalizedRecoverySuggestion=The requested file name is already in use. Try a different file name or location., NSLocalizedDescription=Cannot Save}
2017-01-15 16:45:49.355282 Synopsis[9426:113504]  [ERROR] Write ErrorError Domain=AVFoundationErrorDomain Code=-11823 "Cannot Save" UserInfo={NSUnderlyingError=0x608000e5f6e0 {Error Domain=NSOSStatusErrorDomain Code=-12412 "(null)"}, NSLocalizedRecoverySuggestion=The requested file name is already in use. Try a different file name or location., NSLocalizedDescription=Cannot Save}
2017-01-15 16:45:49.356847 Synopsis[9426:113504] [General] An uncaught exception was raised

2017-01-15 16:45:49.356860 Synopsis[9426:113691] [General] An uncaught exception was raised

2017-01-15 16:45:49.356874 Synopsis[9426:113504] [General] *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
2017-01-15 16:45:49.356879 Synopsis[9426:113691] [General] *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]
2017-01-15 16:45:49.357194 Synopsis[9426:113691] [General] (
	0   CoreFoundation                      0x00007fff961f2e7b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fffaadddcad objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff960efaf0 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 352
	3   CoreFoundation                      0x00007fff960ef95b +[NSDictionary dictionaryWithObjects:forKeys:count:] + 59
	4   Synopsis                            0x000000010002a141 __39-[AppDelegate enqueueFileForTranscode:]_block_invoke + 321
	5   Synopsis                            0x00000001000314e4 -[BaseTranscodeOperation main] + 244
	6   Synopsis                            0x000000010000df20 -[AnalysisAndTranscodeOperation main] + 960
	7   Foundation                          0x00007fff97b9d6e4 -[__NSOperationInternal _start:] + 672
	8   Synopsis                            0x00000001000313e4 -[BaseTranscodeOperation start] + 324
	9   Foundation                          0x00007fff97b9959b __NSOQSchedule_f + 201
	10  libdispatch.dylib                   0x0000000102be7f5c _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x0000000102bffa59 _dispatch_queue_serial_drain + 205
	12  libdispatch.dylib                   0x0000000102bf1489 _dispatch_queue_invoke + 1174
	13  libdispatch.dylib                   0x0000000102bea1d7 _dispatch_root_queue_drain + 671
	14  libdispatch.dylib                   0x0000000102be9ee8 _dispatch_worker_thread3 + 114
	15  libsystem_pthread.dylib             0x0000000102c5f89a _pthread_wqthread + 1299
	16  libsystem_pthread.dylib             0x0000000102c5f375 start_wqthread + 13
)
2017-01-15 16:45:49.357216 Synopsis[9426:113504] [General] (
	0   CoreFoundation                      0x00007fff961f2e7b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fffaadddcad objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff960efaf0 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 352
	3   CoreFoundation                      0x00007fff960ef95b +[NSDictionary dictionaryWithObjects:forKeys:count:] + 59
	4   Synopsis                            0x000000010002a141 __39-[AppDelegate enqueueFileForTranscode:]_block_invoke + 321
	5   Synopsis                            0x00000001000314e4 -[BaseTranscodeOperation main] + 244
	6   Synopsis                            0x000000010000df20 -[AnalysisAndTranscodeOperation main] + 960
	7   Foundation                          0x00007fff97b9d6e4 -[__NSOperationInternal _start:] + 672
	8   Synopsis                            0x00000001000313e4 -[BaseTranscodeOperation start] + 324
	9   Foundation                          0x00007fff97b9959b __NSOQSchedule_f + 201
	10  libdispatch.dylib                   0x0000000102be7f5c _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x0000000102bffa59 _dispatch_queue_serial_drain + 205
	12  libdispatch.dylib                   0x0000000102bf1489 _dispatch_queue_invoke + 1174
	13  libdispatch.dylib                   0x0000000102bea1d7 _dispatch_root_queue_drain + 671
	14  libdispatch.dylib                   0x0000000102be9ee8 _dispatch_worker_thread3 + 114
	15  libsystem_pthread.dylib             0x0000000102c5f89a _pthread_wqthread + 1299
	16  libsystem_pthread.dylib             0x0000000102c5f375 start_wqthread + 13
)
2017-01-15 16:45:49.357402 Synopsis[9426:113691] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff961f2e7b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fffaadddcad objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff960efaf0 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 352
	3   CoreFoundation                      0x00007fff960ef95b +[NSDictionary dictionaryWithObjects:forKeys:count:] + 59
	4   Synopsis                            0x000000010002a141 __39-[AppDelegate enqueueFileForTranscode:]_block_invoke + 321
	5   Synopsis                            0x00000001000314e4 -[BaseTranscodeOperation main] + 244
	6   Synopsis                            0x000000010000df20 -[AnalysisAndTranscodeOperation main] + 960
	7   Foundation                          0x00007fff97b9d6e4 -[__NSOperationInternal _start:] + 672
	8   Synopsis                            0x00000001000313e4 -[BaseTranscodeOperation start] + 324
	9   Foundation                          0x00007fff97b9959b __NSOQSchedule_f + 201
	10  libdispatch.dylib                   0x0000000102be7f5c _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x0000000102bffa59 _dispatch_queue_serial_drain + 205
	12  libdispatch.dylib                   0x0000000102bf1489 _dispatch_queue_invoke + 1174
	13  libdispatch.dylib                   0x0000000102bea1d7 _dispatch_root_queue_drain + 671
	14  libdispatch.dylib                   0x0000000102be9ee8 _dispatch_worker_thread3 + 114
	15  libsystem_pthread.dylib             0x0000000102c5f89a _pthread_wqthread + 1299
	16  libsystem_pthread.dylib             0x0000000102c5f375 start_wqthread + 13
)
libc++abi.dylib: terminating with uncaught exception of type NSException
2017-01-15 16:45:49.357487 Synopsis[9426:113504] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff961f2e7b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fffaadddcad objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff960efaf0 -[__NSPlaceholderDictionary initWithObjects:forKeys:count:] + 352
	3   CoreFoundation                      0x00007fff960ef95b +[NSDictionary dictionaryWithObjects:forKeys:count:] + 59
	4   Synopsis                            0x000000010002a141 __39-[AppDelegate enqueueFileForTranscode:]_block_invoke + 321
	5   Synopsis                            0x00000001000314e4 -[BaseTranscodeOperation main] + 244
	6   Synopsis                            0x000000010000df20 -[AnalysisAndTranscodeOperation main] + 960
	7   Foundation                          0x00007fff97b9d6e4 -[__NSOperationInternal _start:] + 672
	8   Synopsis                            0x00000001000313e4 -[BaseTranscodeOperation start] + 324
	9   Foundation                          0x00007fff97b9959b __NSOQSchedule_f + 201
	10  libdispatch.dylib                   0x0000000102be7f5c _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x0000000102bffa59 _dispatch_queue_serial_drain + 205
	12  libdispatch.dylib                   0x0000000102bf1489 _dispatch_queue_invoke + 1174
	13  libdispatch.dylib                   0x0000000102bea1d7 _dispatch_root_queue_drain + 671
	14  libdispatch.dylib                   0x0000000102be9ee8 _dispatch_worker_thread3 + 114
	15  libsystem_pthread.dylib             0x0000000102c5f89a _pthread_wqthread + 1299
	16  libsystem_pthread.dylib             0x0000000102c5f375 start_wqthread + 13
)
```

--- a/Synopsis/Synopsis/AnalysisAndTranscodeOperation.h
+++ b/Synopsis/Synopsis/AnalysisAndTranscodeOperation.h
@@ -17,8 +17,12 @@
 @property (copy) NSMutableArray* analyzedAudioSampleBufferMetadata;
 @property (copy) NSDictionary* analyzedGlobalMetadata;
 
+@property (assign) BOOL succeeded;
+
 - (instancetype) initWithSourceURL:(NSURL*)sourceURL destinationURL:(NSURL*)destinationURL transcodeOptions:(NSDictionary*)transcodeOptions NS_DESIGNATED_INITIALIZER;
 
 - (instancetype) init NS_UNAVAILABLE;
+
+
 
 @end

--- a/Synopsis/Synopsis/AnalysisAndTranscodeOperation.m
+++ b/Synopsis/Synopsis/AnalysisAndTranscodeOperation.m
@@ -1066,12 +1066,13 @@
         
         // dealloc any analyzers now
         self.availableAnalyzers = nil;
-        
+		
+		self.succeeded = YES;
         [[LogController sharedLogController] appendSuccessLog:@"Finished Analysis Operation"];
     }
     else
     {
-        [[LogController sharedLogController] appendErrorLog:@"Unable to start transcode:"];
+        [[LogController sharedLogController] appendErrorLog:[NSString stringWithFormat:@"Unable to start transcode from %@ to %@:", self.sourceURL, self.destinationURL]];
         if(self.transcodeAssetReader.error)
         {
             [[LogController sharedLogController] appendErrorLog:[@"Read Error" stringByAppendingString:self.transcodeAssetReader.error.debugDescription]];
@@ -1080,6 +1081,7 @@
         {
             [[LogController sharedLogController] appendErrorLog:[@"Write Error" stringByAppendingString:self.transcodeAssetWriter.error.debugDescription]];
         }
+		self.succeeded = NO;
     }
 }
 

--- a/Synopsis/Synopsis/AppDelegate.m
+++ b/Synopsis/Synopsis/AppDelegate.m
@@ -302,34 +302,36 @@ static NSTimeInterval start;
                                 {
                                     // Retarded weak/strong pattern so we avoid retain loopl
                                     __strong AnalysisAndTranscodeOperation* strongAnalysis = weakAnalysis;
-                                    
-                                    NSDictionary* metadataOptions = @{kSynopsisAnalyzedVideoSampleBufferMetadataKey : strongAnalysis.analyzedVideoSampleBufferMetadata,
-                                                                      kSynopsisAnalyzedAudioSampleBufferMetadataKey : strongAnalysis.analyzedAudioSampleBufferMetadata,
-                                                                      kSynopsisAnalyzedGlobalMetadataKey : strongAnalysis.analyzedGlobalMetadata
-                                                                      };
-                                    
-                                    MetadataWriterTranscodeOperation* pass2 = [[MetadataWriterTranscodeOperation alloc] initWithSourceURL:destinationURL destinationURL:destinationURL2 metadataOptions:metadataOptions];
-                                    
-                                    pass2.completionBlock = (^(void)
-                                                             {
-                                                                 [[LogController sharedLogController] appendSuccessLog:@"Finished Analysis"];
-                                                                 
-                                                                 // Clean up
-                                                                 NSError* error;
-                                                                 if(![[NSFileManager defaultManager] removeItemAtURL:destinationURL error:&error])
-                                                                 {
-                                                                     [[LogController sharedLogController] appendErrorLog:[@"Error deleting temporary file: " stringByAppendingString:error.description]];
-                                                                 }
-                                                                 
-                                                                 
-                                                             });
-                                    
-                                    [self.metadataQueue addOperation:pass2];
-                                    
+									
+									if (strongAnalysis.succeeded) {
+										NSDictionary* metadataOptions = @{kSynopsisAnalyzedVideoSampleBufferMetadataKey : strongAnalysis.analyzedVideoSampleBufferMetadata,
+																		  kSynopsisAnalyzedAudioSampleBufferMetadataKey : strongAnalysis.analyzedAudioSampleBufferMetadata,
+																		  kSynopsisAnalyzedGlobalMetadataKey : strongAnalysis.analyzedGlobalMetadata
+																		  };
+										
+										MetadataWriterTranscodeOperation* pass2 = [[MetadataWriterTranscodeOperation alloc] initWithSourceURL:destinationURL destinationURL:destinationURL2 metadataOptions:metadataOptions];
+										
+										pass2.completionBlock = (^(void)
+																 {
+																	 [[LogController sharedLogController] appendSuccessLog:@"Finished Analysis"];
+																	 
+																	 // Clean up
+																	 NSError* error;
+																	 if(![[NSFileManager defaultManager] removeItemAtURL:destinationURL error:&error])
+																	 {
+																		 [[LogController sharedLogController] appendErrorLog:[@"Error deleting temporary file: " stringByAppendingString:error.description]];
+																	 }
+																	 
+																	 
+																 });
+										
+										[self.metadataQueue addOperation:pass2];
+									}
+									
                                 });
-    
+	
     [[LogController sharedLogController] appendVerboseLog:@"Begin Transcode and Analysis"];
-    
+	
     [self.transcodeQueue addOperation:analysis];
 }
 
@@ -340,21 +342,21 @@ static NSTimeInterval start;
     if(fileURLArray)
     {
         start = [NSDate timeIntervalSinceReferenceDate];
-        
+		
         for(NSURL* url in fileURLArray)
         {
             [self enqueueFileForTranscode:url];
         }
-        
+		
         NSBlockOperation* blockOp = [NSBlockOperation blockOperationWithBlock:^{
             NSTimeInterval delta = [NSDate timeIntervalSinceReferenceDate] - start;
-            
+			
             [[LogController sharedLogController] appendSuccessLog:[NSString stringWithFormat:@"Batch Took : %f seconds", delta]];
-            
+			
         }];
-        
+		
         [blockOp addDependency:[self.transcodeQueue.operations lastObject]];
-        
+		
         [self.transcodeQueue addOperation:blockOp];
     }
 }

--- a/Synopsis/Synopsis/MetadataWriterTranscodeOperation.m
+++ b/Synopsis/Synopsis/MetadataWriterTranscodeOperation.m
@@ -793,7 +793,7 @@
     }
     else
     {
-        [[LogController sharedLogController] appendErrorLog:@"Unable to start transcode:"];
+		[[LogController sharedLogController] appendErrorLog:[NSString stringWithFormat:@"Unable to start transcode from %@ to %@:", self.sourceURL, self.destinationURL]];
         [[LogController sharedLogController] appendErrorLog:[@"Read Error" stringByAppendingString:self.transcodeAssetReader.error.debugDescription]];
         [[LogController sharedLogController] appendErrorLog:[@"Write Error" stringByAppendingString:self.transcodeAssetWriter.error.debugDescription]];
     }

